src/Marten/Events/EventGraph.cs

```diff
@@ -155,46 +156,36 @@ public void MapEventType(Type eventType, string eventTypeName)
             JsonTransformation jsonTransformation = null
         )
         {
-            var eventMapping = EventMappingFor(eventType);
+            var eventMapping = typeof(EventMapping<>).CloseAndBuildAs<EventMapping>(this, eventType);
```

Comment:
@jeremydmiller, this is the most important change. I'm setting the mapping here for the event type, to map the old event type name to the new CLR type. The mapping will looks like that now:
- old event type name => new CLR type,
- new event type name => new CLR type,
- new CLR type => new event type name

Before that change, it was:
- old event type name => new CLR type,
- new CLR type => old event type name

Because of that, it was hard (or almost impossible) to have different upcasting logic for different schema versions.

Reply:
That makes sense.